### PR TITLE
Resource add

### LINKED ISSUE #5096: [UI-REWRITE]: Add Resources

### PR DESCRIPTION
**Pull request by gcgoncalves**
# ✨ Feature / Enhancement PR

## 🔗 Epic / Issue

_Link to the epic or parent issue:_
Closes #5096 

---

## 🚀 Summary (1-2 sentences)

Add the resource form functionality (for create only).

### CODE REVIEW

**Review by marekdano — COMMENTED** [review]
### ❌ Discrepancies between the PR and design 

https://www.figma.com/design/H2Dr9QPvRVxCe92V8URvGT/Refiring?node-id=4110-28113&t=zXlvyeP4nEg4pYnv-0

  #### 1. Form header — icon and subtitle missing

  **Design:** Pink/magenta icon (`#ff5aff` background + box/cube icon) + bold "Add resources" heading + subtitle:
  > "Register an URI, upload a file, or paste content to expose it as a resource."

  **PR:** Plain `<h2>Add Resource</h2>`, no icon, no subtitle.

  Fix: same structure as `ToolForm` — coloured icon square + heading + `<p>` description.

  ---

  #### 2. Field order is swapped

  **Design:** Name → URI → Description → MIME type → Content → Visibility → Tags

  **PR:** URI → Name → Description → MIME type → Content → Tags

  Name must come before URI.

  ---

  #### 3. MIME type must be a combobox/select, not a text input

  **Design:** Dropdown/combobox with chevron, defaults to `text/markdown`, **required** (has `*`).

  **PR:** Plain `<Input type="text">`, optional, no asterisk.

  The Figma annotation says *"detect from URI, likely use shadcn combobox"*. Field must be required with validation.

  ---

  #### 4. Visibility field is completely absent

  **Design:** Required select field (default: `Public`) between Content and Tags.

  **PR:** Not implemented at all — field is missing from both `ResourceForm` and `useResourceForm`.

  ---

  #### 5. Content must be a code editor, not a plain textarea

  **Design:** Code editor with line numbers, JetBrains Mono 12px — same style as the
  `inputSchema`/`outputSchema` editors in `ToolForm`.

  **PR:** Plain `<Textarea rows={6}>`.

  ---

  #### 6. Submit button label and alignment

  **Design:** Right-aligned — `Cancel` (ghost) then `Add resources` (black filled).

  **PR:** Left-aligned — `Create Resource` then `Cancel`.

  Both the label and the alignment are wrong.

  ---

  #### 7. X close button should not exist

  **Design:** No X button in the form header; the only dismiss path is the bottom `Cancel` button.

  **PR:** Ghost X button (`aria-label="Close form"`) in top-right of header.

  Remove it.

  ---

  #### 8. Description field should have no label

  **Design:** No label above the description textarea — only the placeholder
  "Add an optional description..."

  **PR:** Explicit `<Label htmlFor="description">Description</Label>` rendered above it.

  ---

  #### 9. Help text below fields is not in the design

  **Design:** No help text under any field.

  **PR:** Every field has a `<p>` help line beneath it (e.g. "Format: resource://gateway/path").

  Remove all help text.

  ---

  #### 10. Form container styling

  **Design:** `rounded-xl` (12px), centered, max-width ~728px.

  **PR:** `rounded-lg`, no max-width, no centering.

  Should align with the `ToolForm` container:
  `mx-auto mt-6 w-full max-w-3xl rounded-xl border shadow-[0_12px_40px_rgba(15,23,42,0.12)]`.

**Comment by gcgoncalves** [comment]
About point 3:

> 3. MIME type must be a combobox/select, not a text input
> Design: Dropdown/combobox with chevron, defaults to text/markdown, required (has *).
> 
> PR: Plain <Input type="text">, optional, no asterisk.
> 
> The Figma annotation says "detect from URI, likely use shadcn combobox". Field must be required with validation.

It is not required on the DB, so I'll leave it as not required. Still, converted it to select. 👍

**Review by marekdano — CHANGES_REQUESTED** [review]
The form is matching the design and looks good!

Some findings: 

**Blocking**

- **Description textarea has no accessible label** (`ResourceForm.tsx`): The comment says "no label, placeholder only" but placeholder text alone fails WCAG 2.1 SC 1.3.1 — screen readers cannot announce the field's purpose. Add a visually-hidden label: `<label className="sr-only" htmlFor="resource-description">Description</label>`.

- **MIME type Select missing `aria-invalid`/`aria-describedby` on error** (`ResourceForm.tsx`): The error paragraph has `id="mime-type-error"` but `<SelectTrigger>` has no `aria-describedby="mime-type-error"` or `aria-invalid={!!errors.mimeType}`. This means screen readers won't associate the error message with the field. Compare with how URI and Name handle this - they set both attributes correctly.

**Suggestions**

- **Remove double sanitization** (`useResourceForm.ts`): `sanitizeString` is called in the Zod `transform` and again inside `getFormData` for the same fields (`uri`, `name`, `description`, `mimeType`, `tags`). Pick one place. The schema transform is the better choice since validation and sanitization happen together.

- **Add URI format validation** (`useResourceForm.ts`): Only `min(1)` is enforced. The placeholder suggests `resource://example/path` but any string passes. Consider adding a `.regex()` or `.url()` refinement to catch obviously malformed URIs before they reach the API.

- **Don't expose `getFormData` on the hook's return value** (`useResourceForm.ts`): It's only used by unit tests, which couples tests to implementation details. Tests should go through `handleSubmit` like real consumers do. Keep `getFormData` as a local function inside the hook.

- **Add delete-flow tests in `Resources.test.tsx`**: The `confirmDelete` handler added to `Resources.tsx` - including the success toast, error toast, and refetch has no test coverage.

- **Add `formatBytes` unit tests** (`utils/format.ts`): The extracted utility has no tests. Edge cases worth covering: `0`, values that produce KB/MB/GB, and negative numbers (currently `Math.log` of a negative returns `NaN`).

File: client/src/components/resources/ResourceForm.tsx [hunk 1]
```diff
@@ -1,154 +1,279 @@
...
+              <SelectTrigger
+                id="resource-visibility"
+                className="w-full rounded-md border-neutral-300 bg-white text-sm text-neutral-900 shadow-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-0 dark:border-neutral-700 dark:bg-neutral-950 dark:text-neutral-100"
+              >
+                <SelectValue />
+              </SelectTrigger>
+              <SelectContent>
+                <SelectItem value="public">Public</SelectItem>
+                <SelectItem value="private">Private</SelectItem>
+                <SelectItem value="team">Team</SelectItem>
```
Comment (marekdano) [inline client/src/components/resources/ResourceForm.tsx]: Every other label goes through `intl.formatMessage`, but these three items are hard-coded English strings.
Add i18n keys for these - the visibility label key already exists in the locale files, just not the option values.

File: client/src/hooks/useResourceForm.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      .optional(),
+    mimeType: z
+      .string()
+      .transform((val) => sanitizeString(val, 200))
+      .optional(),
+    tags: z
+      .string()
+      .transform((val) => sanitizeString(val, 500))
+      .optional(),
+  });
```
Comment (marekdano) [inline client/src/hooks/useResourceForm.ts:21]: `sanitizeString` runs inside the Zod schema transforms and again inside `getFormData()`. They're independent reads from the raw state, so neither uses the other's output. I suggest picking one site.

**Review by marekdano — COMMENTED** [review]
Looks good to merge with a couple of small fixes below.

**Review by marekdano — APPROVED** [review]
Looks good now!

LGTM 🚀  after CI tests go green

**Review by marekdano — APPROVED** [review]
